frame/bags-list/src/benchmarks.rs

```diff
@@ -135,9 +135,41 @@ frame_benchmarking::benchmarks! {
 		);
 	}
 
-	impl_benchmark_test_suite!(
-		Pallet,
-		crate::mock::ExtBuilder::default().build(),
-		crate::mock::Runtime,
-	)
+	put_in_front_of {
+		// The most expensive case for `put_in_front_of`:
+		//
+		// - both heavier's `prev` and `next` are nodes that will need to be read and written.
+		// - `lighter` is the bag's `head`, so the bag will need to be read and written.
```

Comment:
isn't a third condition for most expensive case is: "`heavier` must be a tail or head, so the remove operation also trigger a bag write`. I know the bag will be cached so the additional cost is marginal but still it is more correct no ?

Reply:
I think this will be less expensive because both the initial `heavier.next` and bag will be cached after the remove

So your scenario will look like
- remove
  - bag read/write
  - heavier read
  - lighter read/write
- insert
 - bag read/write (cached)
 - ligher read/write (cached)
 - heavier write
 - total 3 write, 3 read, 2 cached-write, 2 cached read

In the current scenario though we have:
- remove
  - heavier read
  - heavier.next read/write
  - heavier.prev read/write
- insert
  - bag read/write
  - lighter read/write
  - heavier write
- total 5 writes, 5 reads (this is also confirmed storage annotations in benchmarks)

Reply:
yes ok, thanks